spec/models/post_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user) do
+    User.new(
+      name: 'Mateo Lane',
+      photo: 'https://unsplash.com/photos/mateopic',
+      bio: 'Comedian, Italian, Stylish, and part time youtuber',
+      posts_counter: 5
+    )
+  end
+
+  let(:post) do
+    Post.new(
+      author: user,
+      title: 'My first post',
+      text: 'This is my first post',
+      comments_counter: 1,
+      likes_counter: 1
+    )
+  end
+
+  it 'is not valid without a title' do
+    post.title = nil
+    expect(post).to_not be_valid
+  end
+
+  it 'is valid with a title string' do
+    post.title = 'My first post'
+    expect(post).to be_valid
+  end
+
+  it 'is valid with a text string' do
+    post.text = 'This is my first post'
+    expect(post).to be_valid
+  end
+
+  it 'is not valid without a text string' do
+    post.text = nil
+    expect(post).to_not be_valid
+  end
+
+  it 'is only valid with a user' do
+    post.author = user
+    expect(post).to be_valid
+  end
+
+  it 'is not valid if comment counter is less than 0' do
+    post.comments_counter = -1
+    expect(post).to_not be_valid
+  end
+
+  it 'is valid if likes counter is 2' do
+    post.likes_counter = 2
+    expect(post).to be_valid
+  end
+
+  it 'updates posts counter after save' do
+    post.save
+    expect(user.posts_counter).to eq(6)
+  end
```

Comment:
- [ ] Great job so far 👏🏻  I can see that the model specs have **Validation** However you should make sure that  Any custom methods in the Post model have a corresponding unit test. If you check your Post model you can see that you did not add a unit test for this method ⬇️ Let's add this.

```ruby
   def check_last_five_comments
     comments.order(created_at: :desc).limit(5)
  end
```